src/utils/fetchClient.ts

```diff
@@ -29,7 +29,10 @@ function request<T>(
   // for a demo purpose we emulate a delay to see if Loaders work
   return wait(300)
     .then(() => fetch(BASE_URL + url, options))
-    .then(response => response.json());
+    .then(response => response.json())
+    .catch(error => {
+      throw new Error(error);
```

Comment:
Consider providing more informative error messages or handling specific error scenarios in the catch block to improve error handling.